src/createEpicMiddleware.js

```diff
@@ -31,12 +31,20 @@ export function createEpicMiddleware(epic, options = defaultOptions) {
   const epicMiddleware = _store => {
     store = _store;
 
+    const vault = (process.env.NODE_ENV === 'production') ? store : {
+      getState: store.getState,
+      dispatch: (action) => {
+        store.dispatch(action);
+        console.warn(`Your Epic "${epic.name || '<anonymous>'}" called store.dispatch directly. This is an anti-pattern.`);
```

Comment:
`epic.name` refers to the root epic in this scope I think? `export function createEpicMiddleware(epic, ...` If so, it'll probably be confusing. You probably had this code originally inside the `::map(epic => {` scope, in which case `epic` refers to the actual epic that did use `store.dispatch`, but you'd have to create a "vault" for each epic, which I think is an ok tradeoff. 😄 

While you're in here, you might change the root epic variable from `epic` to `rootEpic` to make it less confusing and prevent our `::map(epic => {` from shadowing it.

Reply:
Ah yes. Maybe I can warn the action.type instead to make the message more helpful?

Reply:
Hmm I moved the vault into the .map but it seems the epic there is the root epic, and the name undefined. Maybe I can use the action.type in the warning message instead?

(I chose "vault" as a name for a store with more restrictions)

Reply:
Ohhh right, I forgot that inside the middleware we don't have access to the individual epics because they were composed with `combineEpics`. I think I'll adjust it to just be generic and not mention which specific epic is to blame. I'm also going to change the message a bit to clarify that we're actually deprecating this (store.dispatch inside epics) and it will be removed in v1.0.0.